### ISSUE TITLE
Stuck xroot transfers with gfal2 and XCache

### ISSUE THREAD

**Issue by vokac** [xrootd/xrootd#1808]
Most probably same issue as described in #1739 ... `xrdcopy` is able to recover / reopen connection (after quite long 120s timeout) while `gfal-copy` gets struck for fraction of transfers. This time DPM is not involved, but I'm just using CERN EOS with our XCache `xcache1.farm.particle.cz` (OSG package `xrootd-5.5.0-1.1.osg36.el7.x86_64` deployed by Slate), e.g.
```sh
[lxplus.cern.ch]~% xrdcopy -f root://xcache1.farm.particle.cz//root://eosatlas.cern.ch//eos/atlas/atlasscratchdisk/SAM/1G file:///tmp/1G.xrdcopy
[lxplus.cern.ch]~% gfal-copy --just-copy -f -t 60 root://xcache1.farm.particle.cz//root://eosatlas.cern.ch//eos/atlas/atlasscratchdisk/SAM/1G file:///tmp/1G.xrdcopy
[lxplus.cern.ch]~% gfal-copy --version
gfal2-util version 1.8.0 (gfal2 2.21.1)
...
```
Full log files from client [xrdcopy.log](http://vokac.web.cern.ch/vokac/tmp/xcache-20221026.xrdcopy.log) and [gfal-copy.log](http://vokac.web.cern.ch/vokac/tmp/xcache-20221026.gfal-copy.log).

**Comment by smithdh** [comment]
Hello.

I hope the causes of this have been understood now; I think it is a combination of things. One issue is that the transfer stalls and the other is question of it being restarting/recovering. I think the relevant difference between the xrdcopy and gfal2 case is that gfal2 typically adds the '?xrd.gsiusrpxy=' in the url. This causes some slightly different processing in the client library and in a roundabout way caused the stream to not be recovered after the timeout. This should be addressed in PR #1824. (The description in that PR doesn't really describe this case, but I think it does also cover the problem here).

The other issue - the transfers stalling - I hope will be addressed with PR #1829.

We'll likely discuss these PR in the xroot team somewhat, before including them or similar changes in a release, and then hopefully this issue (1808, and also 1739) should be fixed.

**Comment by abh3** [comment]
I think this is the same issue as in 1813 which has been closed. So, we are going to close this as well. Please reopen with more information, if need be.
